### DISCUSSION TITLE
Alternative designs

### DISCUSSION THREAD

**Discussion by mebs** [mebs/CM4-NAS#5] [Ideas]
The base design was guided by a few leading principles:

1. The board should be as simple as possible to limit room for mistakes.
2. The board should be as easy to assemble by hand as possible.
3. The board should be compact enough to have the case design mostly dictated by the 3.5" hard drives and the fan.

Whether you want to ditch those or not, there's always different ways to make something work, so ideas are welcome!

**Comment by mebs** [comment]
Here are a few alternative designs I keep in mind for later, if the need arises:

- Make the board smaller to fit 2.5" drives. Could be relatively easily done by using SMD components and/or using the back of the board and the space under the CM4 module.
- Have the power delivery right on the board instead of relying on external boards. The back of the base design is mostly unused so it could be used for that.
- We could have the SATA controller card baked in the main board and use one SAS connector on the board instead of 4 SATA connectors. This one looks like quite the adventure....
- It would be possible to keep the CM4 carrier board very small by exporting most of the IO ports to a second (or more) daughter board that would be connected to the main board using a ribbon cable. That would help make the design much more flexible for space constrained applications and it would make exposing the IO more straightforward.

**Comment by olvint** [comment]
I started my own design and i can not finish it. I have very small experience in EasyEda and no experience in KiCAD at all. Routing differential lines in EasyEDA is very hard, and adjusting length is not trully possible.
I made initial design and shared it here - [EasyEDA-oshwlab](https://oshwlab.com/OlegVint/rpi-cm4-nas)
I have following ideas that i used in design. 
1) Size of PCB about 10*75 mm and can be reduced by 10 mm each.
2) All connections should be from one side for proper design of the case and comfort use.
3) I placed PCIE from other side so that SATA ports will face that same side as connectors. And also one can use angled PCIe for more slim fit. [Ali-Link](https://aliexpress.ru/item/4000594216710.html?spm=a2g0o.productlist.0.0.66a9cf0aR3GpoL&algo_pvid=d578f3a7-3f1b-4ee4-8d0f-c6fede03fa11&algo_expid=d578f3a7-3f1b-4ee4-8d0f-c6fede03fa11-8&btsid=0b8b15d416078927898503338e6d6c&ws_ab_test=searchweb0_0,searchweb201602_,searchweb201603_&sku_id=10000003679163905)
3) Full sized HDMI. If a person can solder high density connector for sure he can solder HDMI port. And it can be used not only for debugging. It can be used as media centre. 
4) Power distribution on RPI IO Board made complicated, using several different IC`s. I used resettable fuses. (optionally you  can control voltage and state of fuse using GPIO)
5) One USB C for powering with 5V and one USB A. Switch inside only for updating firmware. Also i think it is needed to make connections to solder inside USB HUB PCB like this - [Ali-Link](https://aliexpress.ru/item/32964934138.html?spm=a2g0o.detail.1000013.1.284d6727UFg2oW&gps-id=pcDetailBottomMoreThisSeller&scm=1007.13339.169870.0&scm_id=1007.13339.169870.0&scm-url=1007.13339.169870.0&pvid=ed3ed027-69ae-4fe2-a4c9-fd2bbb2edb73&_t=gps-id:pcDetailBottomMoreThisSeller,scm-url:1007.13339.169870.0,pvid:ed3ed027-69ae-4fe2-a4c9-fd2bbb2edb73,tpp_buckets:21387%230%23183380%230&_ga=2.52746536.1278776975.1607892437-458060365.1607467585&sku_id=66545473067)
6) MicroSD card. There are CM4 modules without eMMC. And routing SD can is not so hard. 
7) All parts and connectors I used  can be bought on LCSC and have most amount in stock and lowest price. All connectors cost 3$. 

Concerning power supply. I like idea of using 3v3 buck. 5 v can be taken from either USB C or a pin inside. For powering HDD`s it is useful to make MOLEX on board and one can either solder MOLEX or solder wires directly. 

Fan. I do not think it is necessary to make a special part for fan. May be add a transistor+resistor+diode to power fan using one of the GPIO. Rpi-config can automatically enable it based on SOC temperature. And also one can make Python script for getting temperature of hard drives. Or make on PCB zone for FAN components it and mark this zone as unnecessary.

Anyway thanks for your effort guys! Consider my post as small suggestion, i am not professional. 
![Screenshot from 2020-12-13 22-52-09](https://user-images.githubusercontent.com/59015286/102024758-03591500-3da5-11eb-930d-01269d4bc144.png)

![Screenshot from 2020-12-13 22-59-20](https://user-images.githubusercontent.com/59015286/102024797-3b605800-3da5-11eb-83e7-ec4c6350ed1e.png)

![Screenshot from 2020-12-13 23-12-572](https://user-images.githubusercontent.com/59015286/102024840-69459c80-3da5-11eb-850d-ef962de1218c.png)

**Reply by mebs** [reply]
Thanks for sharing, Oleg.
I highly encourage you to try KiCad, I hadn't heard of it before this project but it was fairly easy to pick up thanks to its vast community.
Here's a great tutorial to start with if you're interested: https://github.com/MalphasWats/hawk

If you want to save time, feel free to fork from my repo and start from there. Differential pairs presets are already setup and so are the layers. 
If you want to keep going with your design, a quick search on the web shows that you may be able to convert your EasyEDA project to a KiCad project : https://www.electronics-lab.com/convert-easyeda-files-kicad-just-one-click/
Your design goals are absolutely viable, just make sure you check how things fit together. I see that inserting a PCIe card may mechanically conflict with your H3 and H2 headers, for instance.

Good luck.

**Reply by olvint** [reply]
> Thanks for sharing, Oleg.
> I highly encourage you to try KiCad, I hadn't heard of it before this project but it was fairly easy to pick up thanks to its vast community.
> Here's a great tutorial to start with if you're interested: https://github.com/MalphasWats/hawk
> Good luck.

Thanks for encouragement! Actually i passed you training course and continued project. 
I decided to make it different. Insted of PCIe slot i want to install mini-PCIe for a specific card [AliLink](https://aliexpress.ru/item/4000893496966.html?spm=a2g0o.productlist.0.0.4dc54a8dNXv0pK&algo_pvid=2a8ca77a-4647-468d-8a28-b1afa165b9f4&algo_expid=2a8ca77a-4647-468d-8a28-b1afa165b9f4-0&btsid=0b8b034a16081467774554733ee7c0&ws_ab_test=searchweb0_0,searchweb201602_,searchweb201603_&sku_id=10000010460492914)
I found in comments that it works in Debian. 
2 HDD`s and software RAID is enough for my home NAS. It also reduces cost.


One question, if you don`t mind.  MiniPCIE slot has 2 levels of voltage 3.3 v and 1.2 V. In datasheet for chip ASM 1061 it is written  "Integrated 3.3V to 1.2V switch regulator"
I hope it means i do not have to supply 1.2 V to miniPCIe slot. 

Best regards, Oleg.

**Reply by mebs** [reply]
The low profile card is a great way to go, I considered it and found this card:
https://www.aliexpress.com/item/32849323827.html?spm=a2g0s.8937460.0.0.6e702e0e16MI17
It was a bit hard to find but that was the only one that could support 4 drives with this form factor.
Ended up going with the full size PCI slot mainly to keep compatibility with a broader set of cards but also because larger cards have better thermal solutions and those controllers get very hot.

To answer your question, my knowledge in this area is actually very limited. I would encourage you to follow the official MiniPCI specs, that should be the safest way to go.

**Comment by olvint** [comment]
I published preliminary version [here](https://github.com/olvint/CM4-NAS-MiniPCIE)

**Reply by mebs** [reply]
This looks great, I had never thought of removing some copper where the wifi chip is, that could actually help.
I have also updated my design this weekend with a full size HDMI connector. After trying to solder the high density connectors I realized it wasn't that hard so I added it and made a few changes.

**Comment by JanHBade** [comment]
for me adding 2x USB.30 or adding a second PCIE Slot would be nice...

for a NAS (USB3.0 or PCIE) AND TV (over PCIE-Card)-Server!

PowerSource can be a PicoPSU (https://www.mini-box.com/PicoPSU-80-WI-32V)
5V,3,3V and 12V in one place, no more hassle...

**Reply by mebs** [reply]
Awesome! Thanks for pointing me to that PSU, I had no idea it existed and although the price is a bit high, it would be great for this kind of projects.

As for the PCIe/USB3 ideas, that would involve adding a PCI switch in both cases (remember that the only way to get USB3 on the CM4 is through a PCI USB controller, like on the Pi4). I think this would add more complexity than needed for this project and change its scope. It would start to resemble more of a general purpose IO breakout board which will most likely come to market soon when the CM4 is more widely available.

**Reply by JanHBade** [reply]
OHH yes you are right... USB3.0 and PCIE are not possible...

but with this thing: https://www.amazon.de/Express-Switch-Multiplier-Expansion-Connector/dp/B073YQ9N1H/ref=pd_sbs_5?pd_rd_w=2Zggr&pf_rd_p=ad79fb78-2eb6-4fd8-b228-cb6e6b4589d9&pf_rd_r=Q9ZX1VNCG3JJXHS8Z8KM&pd_rd_r=7909fab9-e61b-4aff-b6db-2d2b2d9c2dd2&pd_rd_wg=zHsEd&pd_rd_i=B073YQ9N1H&psc=1
you have 3 PCIE Slots from 1

the pico psu starts at 25$ https://www.mini-box.com/picoPSU-80

or if size doesn't matter take a normal atx psu...

**Comment by konkav-js** [comment]
I had an idea, that you could put a sata controller on board, wire directly on to the pcie lane, have sata connectors, but it looks like all sata controllers I could find were QFN. I never soldered QFN by hand, and the JLC has none of them in their pcba.

**Reply by mebs** [reply]
There's this other discussion here where this idea was listed: https://github.com/mebs/CM4-NAS/discussions/4
That would be really cool but comes with a lot more complexity and would make the board a lot more expensive. There exists another project out there, the Wiretrustee board that does exactly that but that's a double sided PCB that will likely sell for more that 100USD.

**Reply by konkav-js** [reply]
Well, if you want to build a NAS, you will need a SATA controller. And if it's not on board, then you need to buy one, and you will roughly be there in terms of price. 

The advantage of integrated SATA controller is, that you don't need the PCIe slot. You also won't have two PCBs perpendicular to each other.
The disadvantage is, that while you can actually buy PCIe SATA controller in quantity of 1 easily, I am yet to find a place to be able to buy a SATA controller chip in quantity of 1. Also, if you have no great experience with high-speed PCB, then if you fail to route the high-speed lines correctly, in integrated SATA case you will have a greater chance of damaging the chip with a repeated solder/unsolder.

**Reply by mebs** [reply]
Yes, it is obviously really cool. 

> you will roughly be there in terms of price

I would be curious to research this more, but there's a reason the Wiretrustee board will sell for more. Adding the SATA controller while keeping this form factor (or close to it) will require a double sided PCB which is considerably more expensive to assemble (JLC PCB doesn't even offer that if I remember correctly, you'd have to go to for a more commercial solution). I didn't check the price mind you, that's according to what the people behind Wiretrustee said but I do believe them based on what I saw.
I bought my PCI SATA controller card for less than $20 and a modern Marvel controller chip alone would cost around $10, add all the components around it and assembly and you're sure to fall in another price bracket entirely.
This projects is the cheap, inexperienced DIY approach, leveraging the reliability and economy of scale of outsourced daughter boards (both for power management and the SATA controller).

> You also won't have two PCBs perpendicular to each other.

You should check @olvint's take on this project. He branched from here and changed the PCB to expect smaller PCI cards that mount M.2 style on the board.
